src/gretel_client/transformers/transformers/bucket.py

```diff
@@ -102,37 +112,31 @@ class Bucket(Transformer):
     def __init__(self, config: BucketConfig):
         super().__init__(config)
         # Expand tuple to list
-        if isinstance(config.buckets, tuple):
-            self.buckets = bucket_tuple_to_list(config.buckets)
-        else:
-            self.buckets = config.buckets
-        self.bucket_labels = config.bucket_labels
+        self.buckets = config.buckets
         if self.buckets is None or len(self.buckets) == 0:
             raise ValueError("Empty buckets not permitted.")
-        if len(self.buckets) - 1 != len(self.bucket_labels):
-            raise ValueError("Number of bucket_labels must match number of buckets.")
         self.lower_outlier_label = (
-            self.bucket_labels[0]
+            self.buckets[0][2]
             if config.lower_outlier_label is None
             else config.lower_outlier_label
         )
         self.upper_outlier_label = (
-            self.bucket_labels[-1]
+            self.buckets[-1][2]
             if config.upper_outlier_label is None
             else config.upper_outlier_label
         )
 
     def _transform(self, value: Union[Number, str]) -> Union[Number, str]:
-        if isinstance(value, float) or isinstance(value, int):
+        try:
             return self._mutate_num(value)
-        else:
+        except (TypeError, ValueError):
             return value
 
-    def _mutate_num(self, value: Union[float, int]) -> Union[float, int, str]:
-        if value < self.buckets[0]:
+    def _mutate_num(self, value: Union[Number, str]) -> Union[Number, str]:
```

Comment:
If this also works on strings the _num is a little misleading.

Reply:
+1